lms/djangoapps/discussion_api/api.py

```diff
@@ -714,7 +730,7 @@ def delete_thread(request, thread_id):
 
     """
     cc_thread, context = _get_thread_and_context(request, thread_id)
-    if can_delete(cc_thread, context):
+    if has_permission(cc_thread, context):
```

Comment:
@wajeeha-khalid The point of my comment is to centralize the implementation details (of can_delete and can_view sharing the same implementation, etc) within the permissions.py file.  

There has been discussion here at edX about our current access control implementation.  And this change made here goes against where we are headed.  We will be writing up a wiki on our access control thoughts.  But for the time-being, if you can choose one of these 2 options, it would go with our latest thoughts on this.  I'm happy to discuss further (or in real-time chat) if it's still unclear of our approach.